### ISSUE TITLE
Formatting of `[]byte` in Query.logSQL() 

### ISSUE THREAD

**Issue by BourgeoisBear** [go-ozzo/ozzo-dbx#61]
I'm not sure what the ins and outs are for everyone here, but a problem I'm encountering with SQL logging in ozzo is that it outputs binary strings as raw bytes.  For certain values, this will corrupt the terminal when I `cat` the log files.  By changing line 87 from

    strings.Replace(string(bs), ...

to

    "0x" + hex.EncodeToString(bs)

my logs are far less troublesome.

Any opinions on making this change to ozzo?

**Comment by kPshi** [comment]
Go for it!
As the comment buttresses it's only logging and one should really not parse it's outcome to an extend that depends on the content of binary fields.

**Comment by BourgeoisBear** [comment]
@kPshi Any opinion on serialization format?  I'm using hex since that's how my query tool displays binary fields.  Base64 would be more compact though.

**Comment by kPshi** [comment]
As you seem to be ambitious: how about extracting the logSQL function into an own field (maybe named SQLFormatFunc) of the Query struct?
That would allow to change the format per project.

Personally I'd prefer hex as default format, though, as that makes bytes more readable (you see e.g. a zero terminated field within binary data easily).
A sophisticated format might be like hexdump -C prints it with non-US-ASCII bytes replaced by periods on the right and hex on the left.

**Comment by BourgeoisBear** [comment]
Does @qiangxue have an opinion on this?

**Comment by kolkov** [comment]
Good point about the raw byte logging — that can definitely corrupt terminal output. Hex-encoding `[]byte` values in `logSQL()` is a sensible fix. Would you be up for submitting a PR for this?

This has been open for a long time. Still relevant? We'll close in a couple of weeks if no response.

**Comment by BourgeoisBear** [comment]
In hindsight, I think strconv.Quote might be a better general-purpose default.  I was using binary primary keys, so hex worked for me, but for other applications, strconv.Quote is probably better.  Give me a few days, and I'll take a stab at an implementation to allow changing out the log serializer for []byte.

**Comment by kolkov** [comment]
That's a solid call — `strconv.Quote` is definitely more universal than hex for general logging. A pluggable serializer would be even better. Happy to review whenever it's ready!

**Comment by BourgeoisBear** [comment]
Just sent a pull request for this one.

**Comment by BourgeoisBear** [comment]
Since `string` can also contain binary gibberish, I changed that to `strconv.Quote()` as well.  If I were going to be fastidious about it, it would probably be good to parameterize the ordinary `string` serializer in addition to the `[]byte` one.  I skipped that in my pull request figuring that it may be *too* fastidious.

**Comment by kolkov** [comment]
Fixed in PR #106 — `[]byte` values now use `strconv.Quote()` by default, and there's a pluggable `LogBinaryFormatter` for custom serialization (hex, base64, etc.). Thanks @BourgeoisBear and @kPshi for the discussion and implementation!
